src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+                    if (companyattrs != null && companyattrs.Length > 0)
+                    {
+                        AssemblyCompanyAttribute company = companyattrs[0] as AssemblyCompanyAttribute;
+                        if (company != null && company.Company != null)
+                        {
+                            Company = company.Company;
+                        }
+                    }
+
+                    //set the description based off the description attribute of the given type.
+                    DescriptionAttribute descattr = (DescriptionAttribute)TypeDescriptor.GetAttributes(type)[typeof(DescriptionAttribute)];
+                    if (descattr != null)
+                    {
+                        this.Description = descattr.Description;
+                    }
+
+                    ToolboxBitmapAttribute attr = (ToolboxBitmapAttribute)TypeDescriptor.GetAttributes(type)[typeof(ToolboxBitmapAttribute)];
+                    if (attr != null)
+                    {
+                        Bitmap = attr.GetImage(type, false) as Bitmap;
+                    }
+
+                    bool filterContainsType = false;
+                    ArrayList array = new ArrayList();
+                    foreach (Attribute a in TypeDescriptor.GetAttributes(type))
+                    {
+                        ToolboxItemFilterAttribute ta = a as ToolboxItemFilterAttribute;
+                        if (ta != null)
+                        {
+                            if (ta.FilterString.Equals(TypeName))
+                            {
+                                filterContainsType = true;
+                            }
+                            array.Add(ta);
+                        }
+                    }
+
+                    if (!filterContainsType)
+                    {
+                        array.Add(new ToolboxItemFilterAttribute(TypeName));
+                    }
+
+                    Filter = (ToolboxItemFilterAttribute[])array.ToArray(typeof(ToolboxItemFilterAttribute));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Locks this toolbox item.  Locking a toolbox item makes it read-only and 
+        /// prevents any changes to its properties.
+        /// </summary>
+        public virtual void Lock()
+        {
+            _locked = true;
+        }
+
+        /// <summary>
+        /// Raises the OnComponentsCreated event. This
+        /// will be called when this <see cref='System.Drawing.Design.ToolboxItem'/> creates a component.
+        /// </summary>
+        [SuppressMessage("Microsoft.Security", "CA2109:ReviewVisibleEventHandlers")] //full trust anyway
+        protected virtual void OnComponentsCreated(ToolboxComponentsCreatedEventArgs args)
+        {
+            if (_componentsCreatedEvent != null)
+            {
+                _componentsCreatedEvent(this, args);
+            }
```

Comment:
Nit: similar cleanup could be done here in the future, e.g. `_componentsCreatedEvent?.Invoke(this, args);`.